stubs/tensorflow/tensorflow/raw_ops.pyi

```diff
@@ -0,0 +1,45 @@
+from _typeshed import Incomplete
+from typing import Literal
+
+from tensorflow import Tensor
+from tensorflow._aliases import TensorCompatible
+
+def Fingerprint(data: TensorCompatible, method: Literal["farmhash64"], name: str | None = None) -> Tensor: ...
+def Snapshot(input: TensorCompatible, name: str | None = None) -> Tensor: ...
+def ResourceApplyAdagradV2(
+    var: Tensor,
+    accum: Tensor,
+    lr: TensorCompatible,
+    epsilon: TensorCompatible,
+    grad: TensorCompatible,
+    use_locking: bool = False,
+    update_slots: bool = True,
+    name: str | None = None,
+) -> None: ...
+def ResourceSparseApplyAdagradV2(
+    var: Tensor,
+    accum: Tensor,
+    lr: TensorCompatible,
+    epsilon: TensorCompatible,
```

Comment:
These two must be scalars, so perhaps we can give narrower type (optional)

Reply:
I don't have code to run this function, so I can't check, but very often the TensorFlow doc says that the input must be a scalar, but anything that can be converted to a scalar will do. So for example it will say that a `float32` is required but `"1.2"` would work. However that behavior isn't consistent, so we can't know without testing.

For example:
```python
import tensorflow as tf
tf.keras.optimizers.Adam(learning_rate="1.0")   # Raises ValueError
tf.keras.optimizers.Adam(epsilon="1.0")   # Works
```

So far I think most of the typing for TensorFlow is using overly lax typing for the input in order to avoid false positives, so I would leave `TensorCompatible`.